docs/src/integrations/exchange.md

```diff
@@ -0,0 +1,183 @@
+generate a Solana keypair using any of our [wallet tools](../wallet-guide/cli.md).
+
+We recommend using a unique deposit account for each of your users.
+
+## Listening for Deposits
+
+When a user wants to deposit SOL into your exchange, instruct them to send a
+transfer to the appropriate deposit address.
+
+### Poll for Blocks
+
+The easiest way to track all the deposit accounts for your exchange is to poll
+for each confirmed block and inspect for addresses of interest, using the
+JSON-RPC service of the Solana api node.
+
+1. To identify which blocks are available, send a [`getConfirmedBlocks` request](../apps/jsonrpc-api.md#getconfirmedblocks),
+passing the last block you have already processed as the start-slot parameter:
+
+```bash
+curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc": "2.0","id":1,"method":"getConfirmedBlocks","params":[5]}' localhost:8899
+
+{"jsonrpc":"2.0","result":[5,6,8,9,11],"id":1}
+```
+Not every slot produces a block, so there may be gaps in the sequence of integers.
+
+2. For each block, request its contents with a [`getConfirmedBlock` request](../apps/jsonrpc-api.md#getconfirmedblock):
+
+```bash
+curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc": "2.0","id":1,"method":"getConfirmedBlock","params":[5, "json"]}' localhost:8899
+
+{"jsonrpc":"2.0","result":{"blockhash":"2WcrsKSVANoe6xQHKtCcqNdUpCQPQ3vb6QTgi1dcE2oL","parentSlot":4,"previousBlockhash":"7ZDoGW83nXgP14vnn9XhGSaGjbuLdLWkQAoUQ7pg6qDZ","rewards":[],"transactions":[{"meta":{"err":null,"fee":5000,"postBalances":[2033973061360,218099990000,42000000003],"preBalances":[2044973066360,207099990000,42000000003],"status":{"Ok":null}},"transaction":{"message":{"accountKeys":["Bbqg1M4YVVfbhEzwA9SpC9FhsaG83YMTYoR4a8oTDLX","47Sbuv6jL7CViK9F2NMW51aQGhfdpUu7WNvKyH645Rfi","11111111111111111111111111111111"],"header":{"numReadonlySignedAccounts":0,"numReadonlyUnsignedAccounts":1,"numRequiredSignatures":1},"instructions":[{"accounts":[0,1],"data":"3Bxs3zyH82bhpB8j","programIdIndex":2}],"recentBlockhash":"7GytRgrWXncJWKhzovVoP9kjfLwoiuDb3cWjpXGnmxWh"},"signatures":["dhjhJp2V2ybQGVfELWM1aZy98guVVsxRCB5KhNiXFjCBMK5KEyzV8smhkVvs3xwkAug31KnpzJpiNPtcD5bG1t6"]}}]},"id":1}
+```
+
+The `preBalances` and `postBalances` fields allow you to track the balance
+changes in every account without having to parse the entire transaction. They
+list the starting and ending balances of each account in
+[lamports](../terminology.md#lamports), indexed to the `accountKeys` list. For
+example, if the deposit address if interest is
+`47Sbuv6jL7CViK9F2NMW51aQGhfdpUu7WNvKyH645Rfi`, this transaction represents a
+transfer of 218099990000 - 207099990000 = 11000000000 lamports = 11 SOL
+
+If you need more information about the transaction type or other specifics, you
+can request the block in binary format, and parse it using either our
+[Rust SDK](https://github.com/solana-labs/solana/tree/master/sdk) or
```

Comment:
Does our Rust SDK support this?

Reply:
yep

Reply:
@mvines hmm I just checked again and I think you're referring to `solana-client` which is not included in `solana-sdk`. We should link to https://docs.rs/solana-client instead. It threw me off initially because `solana-sdk` has a client too but with more limited api support

Reply:
Oh, yes.   The `solana-sdk` crate doesn't directly support that.  Need the `solana-client` crate too, as well as the `solana-transaction-status` crate.    The "Rust SDK" is a little messy right now

Reply:
Better to just link to the top-level repo as the "Rust SDK"?